### ISSUE TITLE
Backup script keeps running continuously

### ISSUE THREAD

**Issue by buckbanzai** [thomasfr/hass-addons#1]
Hello,
It seems like something is causing the add-on's backup script to run continuously, it keeps outputting this, and doesn't seem to ever stop on its own without me manually stopping it. I don't think this is expected behavior, so wanted to submit this issue.

Home Assistant OS 8.1
Supervisor supervisor-2022.05.2

Here's a snippet of the start of what this looks like in the log (but it appears to continue on like this forever):
```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun amazon-s3-backup (no readiness notification)
s6-rc: info: service legacy-services successfully started
[08:31:37] INFO: Starting Amazon S3 Backup...
[08:31:39] INFO: Will delete all the oldest local backups except the '3' newest ones.
[08:31:39] INFO: Finished Amazon S3 Backup.
[08:31:39] INFO: Starting Amazon S3 Backup...
[08:31:40] INFO: Will delete all the oldest local backups except the '3' newest ones.
[08:31:40] INFO: Finished Amazon S3 Backup.
[08:31:40] INFO: Starting Amazon S3 Backup...
[08:31:42] INFO: Will delete all the oldest local backups except the '3' newest ones.
[08:31:42] INFO: Finished Amazon S3 Backup.
[08:31:42] INFO: Starting Amazon S3 Backup...
```

**Comment by thomasfr** [comment]
Hey, thanks for giving it out a try and taking time for creating the issue. 
Can you share a screenshot of your backup configuration, automation, and addon screen of the S3 Backup (Please make sure that you do not share any secrets)? That would be very helpful.

**Comment by buckbanzai** [comment]
Here it is, thanks!

Backup Automation:
```
alias: Create a full backup every Friday at 4am
trigger:
  - platform: time
    at: '04:00:00'
condition:
  - condition: time
    before: '00:00:00'
    weekday:
      - fri
    after: '00:00:00'
action:
  - service: hassio.backup_full
    data:
      name: >-
        {{as_timestamp(trigger.now)|timestamp_custom('%Y%m%d%H%M',
        true)}}_automated-backup
```

Upload Automation:
```
alias: Upload to S3
trigger:
  - platform: time
    at: '05:00:00'
action:
  - service: hassio.addon_start
    data:
      addon: 24842e99_amazon-s3-backup
```

Add-on Info:
<img width="1034" alt="Screen Shot 2022-05-25 at 10 24 45 AM" src="https://user-images.githubusercontent.com/553265/170326210-eb1e90eb-68e5-418d-aa8b-1bddfe24ada1.png">

Add-on Configuration:
![Screen Shot 2022-05-25 at 10 26 27 AM](https://user-images.githubusercontent.com/553265/170326612-8369db6b-b5fd-45da-9e37-3417583dc91f.png)

**Comment by ohheyrj** [comment]
I'm seeing similar behaviour

**Comment by thomasfr** [comment]
I am sorry for not coming back earlier. For some reason i did not notice the notifications :( 
The configuration looks good. I am using different bucket region, but that should not be a problem. 
This is what the protocol looks like for me:
<img width="1041" alt="Screenshot 2022-07-25 at 15 38 06" src="https://user-images.githubusercontent.com/287432/180790950-45f347e0-1f55-40bb-9ae1-55e2e4b8a4e7.png">

Could you post your protocol maybe? Make sure to redact any possible confidential information (like s3 bucket).

**Comment by thomasfr** [comment]
Weird, just saw the protocol again you posted in your initial issue.
Automation also looks ok for me at first look.
For reference, these are my automations:

```
- id: backup_create_full_backup
  alias: Create a full backup every day
  trigger:
    - platform: time
      at: 04:00
  action:
    - service: hassio.backup_full
      data:
        name: "{{as_timestamp(trigger.now)|timestamp_custom('%Y%m%d%H%M', true)}}_automated-backup"
- id: backup_upload_to_s3
  alias: Upload to S3
  trigger:
    - platform: time
      at: 04:30
  action:
    service: hassio.addon_start
    data:
      addon: 1244bd6e_amazon-s3-backup
```

**Comment by HarryEMartland** [comment]
I changed the docker file to run the service script directly and it worked as I think is expected (the addon exits). I think the script is being continuously run as it expects the service to be continuously running. 

TLDR: add `CMD [ "/run.sh" ]` to the docker file and move the script out of the services directory. Happy to do a PR if it helps?

**Comment by langtind** [comment]
Same behaviour for me running HA 2022.11.2.

**Comment by twratl** [comment]
Same on the following version. Pretty much makes this unusable.

 Home Assistant 2023.1.7
Supervisor 2022.12.1
Operating System 9.4
Frontend 20230110.0 - latest

**Comment by atrepca** [comment]
Same here, on version:
> Home Assistant 2023.2.5
Supervisor 2023.01.1
Operating System 9.5
Frontend 20230202.0 - latest

@HarryEMartland's fix should work, are you open to PRs?

**Comment by thomasfr** [comment]
Hey, thanks for giving it a try and sorry that you having issues with it. 
Yes I would appreciate a PR. I tried to reproduce it on my side but I couldn’t. It still works for me without any issues. So, yes! Would love to get PRs with a fix

**Comment by thomasfr** [comment]
Can someone explain to me why it’s working for me but not for all of you? @atrepca @HarryEMartland ?

**Comment by atrepca** [comment]
Maybe it's due to the way Home Assistant was installed ([docs](https://www.home-assistant.io/installation/))? I'm running on Raspberry Pi using the recommended installation method - Home Assistant Operating System which has Supervisor for managing HA Core & Add-ons. How is everyone else running HA?

**Comment by atrepca** [comment]
Forked the repo, added `CMD` and the add-on works fine now, here's my changes: https://github.com/thomasfr/hass-addons/compare/main...atrepca:hass-addons:main?expand=1.

The Dockerfile doesn't have `CMD` or `ENTRYPOINT`, and at least one of them needs to be present, according to [the docs](https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact) - maybe this is causing it?
> Dockerfile should specify at least one of CMD or ENTRYPOINT commands.

Just in case it's needed, add-on logs from the successful run from my automation last night:
```bash
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun amazon-s3-backup (no readiness notification)
s6-rc: info: service legacy-services successfully started
[02:30:01] INFO: Starting Amazon S3 Backup...
[02:30:01] INFO: Starting Amazon S3 Backup...
upload: ../../../../backup/a042b24b.tar to s3://<removed>/a042b24b.tar
upload: backup/a042b24b.tar to s3://<removed>/a042b24b.tar
[02:30:45] INFO: Will delete all the oldest local backups except the '3' newest ones.
s6-rc: info: service legacy-services: stopping
s6-rc: info: service legacy-services successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**Comment by thomasfr** [comment]
Fixed in v1.2.2

**Comment by atrepca** [comment]
Confirming version 1.2.2 works as expected, thanks @thomasfr!
